web/src/features/fireWeather/components/tables/StationComparisonTable.tsx

```diff
@@ -0,0 +1,482 @@
+import React from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import Paper from '@material-ui/core/Paper'
+import Typography from '@material-ui/core/Typography'
+import Table from '@material-ui/core/Table'
+import TableBody from '@material-ui/core/TableBody'
+import TableCell from '@material-ui/core/TableCell'
+import TableContainer from '@material-ui/core/TableContainer'
+import TableRow from '@material-ui/core/TableRow'
+import TableHead from '@material-ui/core/TableHead'
+import ToolTip from '@material-ui/core/Tooltip'
+import { DateTime } from 'luxon'
+import { GeoJsonStation } from 'api/stationAPI'
+import { ObservedValue } from 'api/observationAPI'
+import { NoonForecastValue } from 'api/forecastAPI'
+import { ModelValue } from 'api/modelAPI'
+import { formatDateInUTC0, formatDateInPST } from 'utils/date'
+import {
+  TEMPERATURE_VALUES_DECIMAL,
+  RH_VALUES_DECIMAL,
+  WIND_SPEED_VALUES_DECIMAL,
+  PRECIP_VALUES_DECIMAL,
+  DEW_POINT_VALUES_DECIMAL
+} from 'utils/constants'
+
+const useStyles = makeStyles({
+  paper: {
+    padding: '5px'
+  },
+  typography: {},
+  lightColumnHeader: {
+    textAlign: 'center',
+    padding: '2px',
+    minWidth: '60px'
+  },
+  lightColumn: {
+    textAlign: 'right',
+    padding: '2px'
+  },
+  windSpeedValue: {
+    whiteSpace: 'nowrap'
+  },
+  relativeHumidityValue: {
+    whiteSpace: 'nowrap'
+  },
+  windDirectionValue: {
+    whiteSpace: 'nowrap'
+  },
+  precipitationValue: {
+    whiteSpace: 'nowrap'
+  },
+  darkColumn: {
+    backgroundColor: '#fafafa',
+    padding: '1px',
+    textAlign: 'right'
+  },
+  darkColumnHeader: {
+    backgroundColor: 'rgb(240, 240, 240)',
+    textAlign: 'center',
+    padding: '2px',
+    minWidth: '60px'
+  }
+})
+
+interface Props {
+  timeOfInterest: string
+  stationCodes: number[]
+  stationsByCode: Record<number, GeoJsonStation | undefined>
+  allNoonForecastsByStation: Record<number, NoonForecastValue[] | undefined>
+  observationsByStation: Record<number, ObservedValue[] | undefined>
+  allHighResModelsByStation: Record<number, ModelValue[] | undefined>
+  allRegionalModelsByStation: Record<number, ModelValue[] | undefined>
+  allModelsByStation: Record<number, ModelValue[] | undefined>
+}
+
+const findNoonMatch = (
+  noonDate: string,
+  collection: ModelValue[] | undefined
+): ModelValue | undefined => {
+  return collection?.find((item: ModelValue) => item.datetime === noonDate)
+}
+
+interface AccumulatedPrecipitation {
+  precipitation: number | undefined
+  modelValues: ModelValue[]
+}
+
+const calculateAccumulatedPrecip = (
+  noonDate: string,
+  collection: ModelValue[] | undefined
+): AccumulatedPrecipitation | undefined => {
+  // We are calculating the accumulated precipitation from 24 hours before noon.
```

Comment:
this is where I'm still fuzzy about the precip calculations, and it might be relevant to the ticket I'm working on now. You're calculating the accum precip for the last 24 hours before noon for a model run (so accum precip for May 4 at noon would be all precip between May 3 1pm - May 4 noon), but human forecasters write forecasts for the accum precip for ... all of May 4? Or just the first 12 hours of May 4?

Reply:
I still need to talk to a forecaster about this - it's super confusing. (I also need to check - I think I'm doing observed wrong here)

In terms of observed, we can only report on the last 24 hours (being as we can't observe the future).
This limitation makes me believe that surely you need to compare forecast and model date for the last 24 hours - otherwise what are you comparing?

Reply:
Yes, this is directly relevant to my current ticket.

I was working with the understanding (really just an assumption) that forecasters make precip forecasts for the total accum precip for the 24-hour period of the forecast day. But on second thoughts that's kind of contradictory to the premise of the rest of their forecast, which is to describe what's going to be happening at noon. So yeah, I think we need to clarify this ASAP.